astropy/cosmology/tests/test_units.py

```diff
@@ -217,3 +217,157 @@ def test_equivalency_context_manager():
     # check starting with only the dimensionless_redshift equivalency.
     assert len(base_registry.equivalencies) == 1
     assert str(base_registry.equivalencies[0][0]) == "redshift"
+
+
+@pytest.mark.skipif(not HAS_SCIPY, reason="Cosmology needs scipy")
+class Test_unsuitable_units:
+#Ensure that unsuitable units are rejected
+
+    def test_Om_units(self):
```

Comment:
My tendency would be to make this easier to look at by doing
```
@pytest.mark.parametrize('method', ['Om', ....])
def test_unit_exception(self, method):
    with pytest.raises(u.UnitConversionError):
        getattr(Planck13, method)([0., 1.] * u.m)
```

Reply:
Agreed, this looks better, I've changed it now

Reply:
Done. Thanks @akeemlh